edx_rest_framework_extensions/utils.py

```diff
@@ -38,21 +52,35 @@ def jwt_decode_handler(token):
 
     options = {
         'verify_exp': api_settings.JWT_VERIFY_EXPIRATION,
+        'verify_aud': settings.JWT_AUTH['JWT_VERIFY_AUDIENCE'],
         'require_exp': True,
         'require_iat': True,
     }
 
-    try:
-        return jwt.decode(
-            token,
-            api_settings.JWT_SECRET_KEY,
-            api_settings.JWT_VERIFY,
-            options=options,
-            leeway=api_settings.JWT_LEEWAY,
-            audience=api_settings.JWT_AUDIENCE,
-            issuer=api_settings.JWT_ISSUER,
-            algorithms=[api_settings.JWT_ALGORITHM]
-        )
-    except jwt.InvalidTokenError:
-        logger.exception('JWT decode failed!')
-        raise
+    # JWT_ISSUERS is not one of DRF-JWT's default settings, and cannot be accessed
+    # using the `api_settings` object without overriding DRF-JWT's defaults.
+    jwt_issuers = get_jwt_issuers()
+    if not jwt_issuers:
+        msg = 'JWT_ISSUERS not found in JWT_AUTH.  Please check the system configuration.'
+        logger.error(msg)
+        raise ImproperlyConfigured(msg)
+    for jwt_issuer in get_jwt_issuers():
+        try:
+            decoded = jwt.decode(
```

Comment:
Bypass the assignment and just return here. You aren't doing anything with the variable.